### PR TITLE
fix README: Update contribution guidelines

### PR DESCRIPTION
**Pull request by wesrobin**
### Changed
- Rewrote README to be more instructive for open source contributors

### CODE REVIEW

**Comment by echarrod** [comment]
Ah sorry Wes, I probably didn't explain this enough - this file you've changed is the one displayed on the profile of Luno org, which we want to keep.

I think this stuff should instead go in a CONTRIBUTING.md file in this repo, which then if any of our repos _don't_ have a CONTRIBUTING.md file of their own specified, this will be used.

Bit more info on https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file

**Comment by wesrobin** [comment]
> Ah sorry Wes, I probably didn't explain this enough - this file you've changed is the one displayed on the profile of Luno org, which we want to keep.
> 
> I think this stuff should instead go in a CONTRIBUTING.md file in this repo, which then if any of our repos _don't_ have a CONTRIBUTING.md file of their own specified, this will be used.
> 
> Bit more info on https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file

Can you tell I don't do open source much lol

File: CONTRIBUTING.md
Comment (wesrobin) [inline CONTRIBUTING.md:5]: Is REPONAME a keyword that will be replaced when this default is shown on a repo? If not, is there maybe such a thing that anyone knows about? It's a bit meh seeing `REPONAME`

Comment (wesrobin) [reply CONTRIBUTING.md:5]: Doesn't really look like it, maybe we can write this a little better like 'Fork the repo' and 'create a PR against the main repository'?

Comment (echarrod) [reply CONTRIBUTING.md:5]: Yes I don't think this is possible unfortunately, might just have to say "the repo"

Comment (wesrobin) [reply CONTRIBUTING.md:5]: Chat gippity said we should use a github action 😂

Comment (wesrobin) [reply CONTRIBUTING.md:5]: Yeah no, I agree, it just seemed like an amusing amount of unneeded effort

File: CONTRIBUTING.md
Comment (echarrod) [inline CONTRIBUTING.md:2]: Maybe we can expand this a bit more?
```suggestion

We welcome contributions from the community! Whether you're fixing a bug, adding a feature, or improving documentation, your contributions help us improve this project for everyone!
```

File: CONTRIBUTING.md
Comment (echarrod) [inline CONTRIBUTING.md:3]: Maybe we should put a part in about discussions actually

```suggestion
## Discussions

If you're unsure about anything, feel free to ask questions by opening a discussion or reaching out to us in the issues section.

## Pull Requests
For pull requests (PRs), please stick to the following guidelines:
```

File: CONTRIBUTING.md
Comment (echarrod) [inline CONTRIBUTING.md:12]:
```suggestion

### Hooks
```

**Review by echarrod — APPROVED** [review]
Looks good! Would just like another pair of eyes from @wenkanglu / @solethus / @neilgarb before we merge, since it'll be used on all repos 😄

File: CONTRIBUTING.md
Comment (andrewwormald) [reply CONTRIBUTING.md:2]: +1

File: CONTRIBUTING.md
Comment (andrewwormald) [reply CONTRIBUTING.md:3]: Yes please

File: CONTRIBUTING.md
Comment (andrewwormald) [reply CONTRIBUTING.md:12]: Do we just want to call it `Pre-Commit Hooks`?

**Comment by andrewwormald** [comment]
Nice work @wesrobin !

File: CONTRIBUTING.md
Comment (echarrod) [reply CONTRIBUTING.md:12]: Originally was thinking we can call it hooks, as I've found it useful to only run tests on pre-push. And so it would really be "pre-commit and pre-push hooks". 

I guess this is only useful for repos where tests take a long time, but might be useful to be a bit more generic here, so that we have freedom to add this hook as a pre-push, if we want to.

**Pull request merged**
